lib/spack/spack/cmd/python.py

```diff
@@ -130,8 +130,7 @@ def python_interpreter(args):
     elif args.python_args:
         propagate_exceptions_from(console)
         sys.argv = args.python_args
-        with open(args.python_args[0]) as file:
-            console.runsource(file.read(), args.python_args[0], "exec")
+        runpy.run_path(args.python_args[0], run_name="__main__")
```

Comment:
If we're using run_path here, probably shouldn't bother constructing the interactive console above.  Also do we need `-u` anymore with this change?  That is a change I made in the other PR.

Reply:
Agreed about the console, the `-u` thing is somewhat orthogonal.  Currently the parallel tests run like this: `spack unit-test --dist loadfile --tx '4*popen//python=./bin/spack-tmpconfig python -u ./bin/spack python' -x` and xdist adds a `-u` flag to the 'spack python' invocation which we fail to parse if it's removed.  How were you invoking to avoid that?  

In an ideal world we could use just `spack unit-test -n auto` or `-n 4` or whatever (these are normal xdist options).  Both your PR and this one get us closer to that, since the workers actually _start_ now without any extra fiddling,  but I haven't figured out a way to get the temporary configuration wrapper or an equivalent to work that way, so they all end up running with the same tmpdirs and failing pretty badly.  If we could somehow get that incorporated or whatever so that the multiprocessing or pytest setup gets us to the same place, it would be pretty trivial to use the parallel test stuff.